src/components/JUPTable/JUPTable.tsx

```diff
@@ -0,0 +1,184 @@
+  const createSortHandler = (property: keyof Data) => (event: React.MouseEvent<unknown>) => {
+    onRequestSort(event, property);
+  };
+
+  return (
+    <TableHead>
+      <TableRow>
+        {headCells.map((headCell) => (
+          <TableCell
+            key={headCell.id}
+            align={headCell.numeric ? "right" : "left"}
+            padding={headCell.disablePadding ? "none" : "normal"}
+            sortDirection={orderBy === headCell.id ? order : false}
+          >
+            <TableSortLabel
+              active={orderBy === headCell.id}
+              direction={orderBy === headCell.id ? order : "asc"}
+              onClick={createSortHandler(headCell.id)}
+            >
+              {headCell.label}
+              {orderBy === headCell.id ? (
+                <Box component="span" sx={visuallyHidden}>
+                  {order === "desc" ? "sorted descending" : "sorted ascending"}
+                </Box>
+              ) : null}
+            </TableSortLabel>
+          </TableCell>
+        ))}
+      </TableRow>
+    </TableHead>
+  );
+};
+
+interface IJUPTableProps {
+  headCells: Array<IHeadCellProps>;
+  rows: Array<React.ReactElement>;
+  children?: Array<React.ReactElement>;
+  title: string;
+}
+
+const JUPTable: React.FC<IJUPTableProps> = ({ children, headCells, rows, title }) => {
+  const [rowsPerPage, setRowsPerPage] = React.useState(3);
+  const [order, setOrder] = React.useState<Order>("asc");
+  const [orderBy, setOrderBy] = React.useState<any>("date");
+  const [page, setPage] = React.useState(0);
+  let emptyRows;
+
+  const handleRequestSort = (event: React.MouseEvent<unknown>, property: keyof Data) => {
+    const isAsc = orderBy === property && order === "asc";
+    setOrder(isAsc ? "desc" : "asc");
+    setOrderBy(property);
+  };
+
+  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  if (rows !== undefined) {
+    // Avoid a layout jump when reaching the last page with empty rows.
+    emptyRows = page > 0 ? Math.max(0, (1 + page) * rowsPerPage - rows.length) : 0;
+  } else {
+    emptyRows = 0;
+  }
+
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  return (
+    <TableBackground>
+      <TableContainer>
+        <TableTitle title={title} />
+        <Table aria-labelledby="tableTitle" size={"small"}>
+          <EnhancedTableHead headCells={headCells} order={order} orderBy={orderBy} onRequestSort={handleRequestSort} rowCount={rows?.length || 0} />
+          <TableBody>
+            {rows?.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)}
+            {emptyRows > 0 && (
+              <TableRow
+                style={{
+                  height: 33 * emptyRows,
+                }}
+              >
+                <TableCell colSpan={6} />
```

Comment:
i'm not sure you'll always have the same number of columns, so this colspan of 6 could present issues in the future fyi. it'll only be visual, so you should spot it pretty easy if it happens.

Reply:
found a way to fix this with `headCells.length` which will probably get changed again when I finish making the table data-agnostic.